python/plugins/processing/algs/qgis/StatisticsByCategories.py

```diff
@@ -90,7 +91,10 @@ def processAlgorithm(self, parameters, context, feedback):
             feedback.setProgress(int(current * total))
             attrs = feat.attributes()
             try:
-                value = float(attrs[value_field_index])
+                if value_field_index > 0:
+                    value = float(attrs[value_field_index])
+                else:
+                    value = 0
```

Comment:
I guess this should be `NULL` instead?

Reply:
I thought that 0 would be safer (and it works fine in my tests), as I was not sure about how qgsstatisticalsummary would handle NULLs.  But whatever works is of course OK.